### PR TITLE
Add facingMode detection

### PR DESCRIPTION
**Pull request by Ocupe**
Take advantage of the new facingMode detection https://github.com/livekit/client-sdk-js/pull/738 to stop mirroring the local participant's camera track when using backward facing mobile or OBS virtual cameras.

### CODE REVIEW

File: packages/react/etc/components-react.api.md [hunk 1]
```diff
@@ -458,7 +458,7 @@ export function useClearPinButton(props: ClearPinButtonProps): {
...
+    className: any;
```
[the remark this answers is not in the document]
Comment (Ocupe) [reply packages/react/etc/components-react.api.md]: probably typescript missmatch, its gone now

File: packages/react/src/hooks/useFacingMode.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  trackReference: TrackReferenceOrPlaceholder,
+): 'user' | 'environment' | 'left' | 'right' | 'undefined' {
+  if (!trackReference.participant.isLocal) {
+    return 'undefined';
+  }
+
+  let mediaStreamTrack: LocalTrack | undefined;
+  if (trackReference.publication === undefined) {
+    if (trackReference.participant.videoTracks.size > 0) {
+      mediaStreamTrack = Array.from(trackReference.participant.videoTracks.values())[0]
```
Comment (lukasIO) [inline packages/react/src/hooks/useFacingMode.ts:22]: what's the idea here?

Comment (Ocupe) [reply packages/react/src/hooks/useFacingMode.ts:22]: If you are alone in the room the the publication piped into the hook is unknown. This is away to still get the local track.

File: packages/react/src/hooks/useFacingMode.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  if (!trackReference.participant.isLocal) {
+    return 'undefined';
+  }
+
+  let mediaStreamTrack: LocalTrack | undefined;
+  if (trackReference.publication !== undefined) {
+    mediaStreamTrack = trackReference.publication.track as LocalTrack;
+  }
+
+  if (mediaStreamTrack) {
```
Comment (lukasIO) [inline packages/react/src/hooks/useFacingMode.ts:24]: nit: the three conditions here could be boiled down to
```
if (ref.publication instanceof LocalTrackPublication)
...
else 
return undefined
```

**Review by lukasIO — APPROVED** [review]

**Pull request merged**